### ISSUE TITLE
Transfer learning on trained model

### ISSUE THREAD

**Issue by Storm112112** [matterport/Mask_RCNN#2767]
Hi, 

I have a question about the retraining model.

I've trained the model with 2 classes using Imagenet pre-trained weights it's working perfectly fine. but, now I need to add a new class to that trained model without using the previous 2 class dataset. can someone please guide me through this?

**Comment by mnnsolanki** [comment]
Your model is trained only on two classes. 

So in binary classification, your model has two neurons in the last layer. That's why it'll  gives a result only for those 2 classes. 

But in order to add one more class, means your architecture of model is also changing. Now you'll have 3 neurons in last layer. It's multi-class image classification problem. 

So the only way is to train it again. 

First, prepare the dataset: 
    - Equal distribution of images across the classes is necessary. 
    - Same goes for train, validation & test. 

Now, train it again. 
    - You need to change the loss function from sigmoid to softmax for the last layer. 
    - Use cross-entropy for backprop.
